src/resource/nonRdfData.ts

```diff
@@ -152,9 +155,10 @@ export async function deleteFile(
     method: "DELETE",
   });
 
-  if (!response.ok) {
-    throw new Error(
-      `Deleting the file at \`${url}\` failed: ${response.status} ${response.statusText}.`
+  if (internal_isUnsuccessfulResponse(response)) {
+    throw new FetchError(
+      `Deleting the file at \`${url}\` failed: ${response.status} ${response.statusText}.`,
```

Comment:
```suggestion
      `Deleting the file at [${url}] failed: [${response.status}] [${response.statusText}].`,
```

Reply:
It seems every instance of `new FetchError()` in this PR is duplicating the same text at the end of the error message string (i.e. `at [${url}] failed: [${response.status}] [${response.statusText}]`). So instead I'd just add a parameter to the  constructor to add it automatically, e.g. 
```
constructor(message: string, errorResponse: Response & { ok: false }, appendDetails:boolean = true) {
  super(`${message}${appendDetails  ? ` Failed at URL [${url}]: [${response.status}] [${response.statusText}].` : ''}`);
```
Messages would need to change accordingly, i.e. this would just become `throw new FetchError("Deleting file.", response);`.